acir/src/native_types/arithmetic.rs

```diff
@@ -149,11 +149,23 @@ impl Expression {
         Ok(expr)
     }
 
-    // TODO: possibly rename, since linear can have one mul_term
     pub fn is_linear(&self) -> bool {
         self.mul_terms.is_empty()
     }
 
+    // Returns a witness belonging to the expression
+    pub fn get_witness(&self) -> Option<Witness> {
+        if self.linear_combinations.is_empty() {
+            if self.mul_terms.is_empty() {
+                None
+            } else {
+                Some(self.mul_terms[0].1)
+            }
```

Comment:
Here's how it would look with both for comparison
```rs
self.linear_combinations.get(0).map(|(_, wit)| *wit)
    .unwrap_or_else(|| self.mul_terms.get(0).map(|(_, wit, _)| *wit))
```

Reply:
Yes, I don't see the point to make something clear into something unreadable. Is it more efficient?

Reply:
I find my version more readable tbh, "I want the first thing in this array and then pull out the witness, otherwise None" Vs "is this thing empty? If so then return None otherwise pull out the witness from first element"

In the first you can just read through it whereas I need to scan up and down for the second.

Reply:
This is just a personal opinion. 
I am not saying my version is better, but that I prefer mine (because I don't have to read an inner lambda function and try to figure out what this code is really doing and how), and you prefer yours because of the reasons you said.

Reply:
Why does this method just take the first element from mul_terms of linear_combinations?

- Is it only called when there is either one mul_term of one linear_combination?
- Is it okay that it is ignoring the coefficient for the witness being returned?

In general, I'd change the naming of this and add a more detailed comment, because get_witness() usually infers that there is a canonical Witness to get. Whereas in general an expression can contain many witnesses

Reply:
> because I don't have to read an inner lambda function and try to figure out what this code is really doing and how

I agree that it is personal preference, though it seems odd to me to avoid lambda functions because you need to read their blocks anymore than you should avoid if statements because you need to read their blocks.

Reply:
Closing as it's mostly a matter of taste.